File: backend/app/models/plot.py
from sqlalchemy import Column, Integer, String, Numeric, CHAR, ForeignKey, TIMESTAMP
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base

class Plot(Base):
    __tablename__ = "plots"

    plot_id = Column(Integer, primary_key=True, index=True)
    society_id = Column(Integer, ForeignKey("society.society_id"), nullable=False)
    plot_code = Column(String(20), unique=True, nullable=False)
    area_sqyd = Column(Numeric(10, 2), nullable=True)
    area_sqft = Column(Numeric(10, 2), nullable=True)
    type = Column(CHAR(1), nullable=True)
    created_at = Column(TIMESTAMP, server_default=func.now())

    society = relationship("Society", back_populates="plots")
    floors = relationship("Floor", back_populates="plot", cascade="all, delete")

File: frontend/src/types/plot.ts
// types/plot.ts
export type Plot = {
  plot_id: number
  plot_code: string
  area_sqyd?: number | null
  area_sqft?: number | null
  type?: string | null
  created_at: string
}

export type CreatePlotPayload = {
  society_id: number
  plot_code: string
  area_sqyd?: number | null
  area_sqft?: number | null
  type?: string | null
}

export type UpdatePlotPayload = {
  area_sqyd?: number | null
  area_sqft?: number | null
  type?: string | null
}
